lib/Doctrine/ORM/Mapping/ReflectionEnumProperty.php

```diff
@@ -68,7 +53,7 @@ public function getValue($object = null)
      * @param object                         $object
      * @param int|string|int[]|string[]|null $value
      */
-    public function setValue($object, $value = null): void
+    public function setValue(mixed $object, mixed $value = null): void
```

Comment:
Doesn't `mixed $value` break with the `getValue` method?

Reply:
`setValue()` is overridden from `ReflectionProperty` which declares `mixed` as parameter type already. I'm only allowed to make parameter types wider when overriding and it can't get wider than `mixed`. So `mixed` is actually the only possible native type here.

Reply:
Then I would expect `getValue()` to return mixed too. https://www.php.net/manual/en/reflectionproperty.getvalue.php

If any value is set that is not part of `int|string|array|null`, then we'll have a TypeError.

Reply:
> If any value is set that is not part of `int|string|array|null`, then we'll have a TypeError.

No, `$enum->value` would crash before that. Our implementation of `getValue()` can only return `int|string|array|null`, we can be sure about that.

Note, I'm declaring the parameter type as `mixed` because of a technical limitation which is PHP not supporting generics natively. The doc block is more precise in that regard.